### PR TITLE
New structure

### PR DESCRIPTION
**Pull request by mabels**

### CODE REVIEW

File: src/blockstore/loader.ts [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+
+interface Startable {
+  start(): Promise<void>;
+}
+
+class EnsureStart<T extends Startable> {
+  readonly oncePerStore = new ResolveOnce<T>();
+
+  reset() {
+    this.oncePerStore.reset();
```
Comment (boorad) [inline src/blockstore/loader.ts:80]: `ResolveOnce` doesn't have a `reset()` function

File: src/blockstore/loader.ts [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+    const toClose = await Promise.all([this.carStore(), this.metaStore(), this.fileStore(), this.remoteWAL()]);
+    await Promise.all(
+      toClose.map((store) => {
+        store.close();
+      }),
+    );
+    this.ensureCarStore.reset();
+    this.ensureFileStore.reset();
+    this.ensureRemoteFileStore.reset();
+    this.ensureMetaStore.reset();
```
Comment (boorad) [inline src/blockstore/loader.ts]: same for these `reset()` calls

**Review by boorad — COMMENTED** [review]
found a couple of issues in loader

File: src/blockstore/remote-wal.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+  async destroy() {
+    await this.ready();
+    await this._destroy();
+  }
+
+  abstract _load(branch?: string): Promise<WALState | Falsy>;
+  abstract _save(state: WALState, branch?: string): Promise<void>;
+  abstract _close(): Promise<void>;
+  abstract _destroy(): Promise<void>;
```
Comment (boorad) [inline src/blockstore/remote-wal.ts:193]: Why the underscores on these methods?

File: src/blockstore/loader.ts (hunk 2 above)
Comment (mabels) [reply src/blockstore/loader.ts]: you need to do an pnpm i

Comment (boorad) [reply src/blockstore/loader.ts]: `pnpm i` was already run.  If it resolves these four, it doesn't resolve the one above on line 80.

**Pull request merged**